src/Uno.UI/DirectUI/DXamlCore.cs

```diff
@@ -1,42 +1,60 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// DXamlCore.h, DXamlCore.cpp
+
+#nullable enable
+
+using System;
+using Uno.UI.Xaml.Controls;
+using Uno.UI.Xaml.Core;
 using Windows.ApplicationModel.Resources;
-using Windows.UI.Xaml;
-using Windows.UI.Xaml.Controls;
+using Windows.Foundation;
 
 namespace DirectUI
 {
 	internal class DXamlCore
 	{
-		private static DXamlCore _current;
-		private ElementSoundPlayerService _elementSoundPlayerServiceNoRef;
-		private BuildTreeService _buildTreeService;
-		private BudgetManager _budgetManager;
+		private static readonly Lazy<DXamlCore> _current = new Lazy<DXamlCore>();
 
-		public static DXamlCore GetCurrent()
-			=> _current ??= new DXamlCore();
+		private BuildTreeService? _buildTreeService;
+		private BudgetManager? _budgetManager;
 
 		// UNO: This should **NOT** create the singleton!
 		//		_but_ if we do return a 'null' the 'OnApplyTemplate' of the `CalendarView` will fail.
 		//		As for now our implementation of the 'DXamlCore' is pretty light and stored as a basic singleton,
 		//		we accept to create it even with the "NoCreate" overload.
```

Comment:
should this comment stay here?

Reply:
I think so, still using singleton, for future it should probably be created differently